src/it/resources/com/google/checkstyle/test/chapter4formatting/rule462horizontalwhitespace/InputParenPad.java

```diff
@@ -4,343 +4,343 @@
-        //  '')' is preceded with whitespace.'
-        String d = ((String) b);
+  private void except() {
+    java.util.ArrayList<Integer> arrlist = new java.util.ArrayList<Integer>( 5 );
+    // 2 violations above:
+    //  ''(' is followed by whitespace.'
+    //  '')' is preceded with whitespace.'
+    arrlist.add( 20); // violation ''(' is followed by whitespace.'
+    arrlist.add(15 ); // violation '')' is preceded with whitespace.'
+    arrlist.add( 30 );
+    // 2 violations above:
+    //  ''(' is followed by whitespace.'
+    //  '')' is preceded with whitespace.'
+    arrlist.add(45);
+    try {
+      ( arrlist ).remove( 2);
+      // 3 violations above:
+      //  ''(' is followed by whitespace.'
+      //  '')' is preceded with whitespace.'
+      //  ''(' is followed by whitespace.'
+    } catch ( IndexOutOfBoundsException x ) {
+      // 2 violations above:
+      //  ''(' is followed by whitespace.'
+      //  '')' is preceded with whitespace.'
+      x.getMessage();
     }
+    org.junit.Assert.assertThat( "123", org.hamcrest.CoreMatchers.is( "123" ) );
+    // 4 violations above:
+    //  ''(' is followed by whitespace.'
+    //  ''(' is followed by whitespace.'
+    //  '')' is preceded with whitespace.'
+    //  '')' is preceded with whitespace.'
+    org.junit.Assert.assertThat( "Help! Integers don't work",
+            // violation above ''(' is followed by whitespace.'
+            0, org.hamcrest.CoreMatchers.is( 1 ) );
+    // 3 violations above:
+    //  ''(' is followed by whitespace.'
+    //  '')' is preceded with whitespace.'
+    //  '')' is preceded with whitespace.'
+  }
 
-    public int something( Object o ) {
-        // 2 violations above:
-        //  ''(' is followed by whitespace.'
-        //  '')' is preceded with whitespace.'
-        if ( o == null || !( o instanceof Float ) ) {
-            // 4 violations above:
-            //  ''(' is followed by whitespace.'
-            //  ''(' is followed by whitespace.'
-            //  '')' is preceded with whitespace.'
-            //  '')' is preceded with whitespace.'
-            return -1;
-        }
-        return Integer.valueOf( 22 ).compareTo( (Integer) o );
-        // 4 violations above:
-        //  ''(' is followed by whitespace.'
-        //  '')' is preceded with whitespace.'
-        //  ''(' is followed by whitespace.'
-        //  '')' is preceded with whitespace.'
+  private void tryWithResources() throws Exception {
+    try (AutoCloseable a = null) { } // ok
+    try (AutoCloseable a = null;
+        AutoCloseable b = null) { } // ok
+    try (AutoCloseable a = null;
+        AutoCloseable b = null) { } // ok
+    try (AutoCloseable a = null;
+        AutoCloseable b = null) { } // ok
+    try (AutoCloseable a = null ) { } // violation '')' is preceded with whitespace.'
+    try (AutoCloseable a = null;
+        AutoCloseable b = null ) { }
+    // violation above '')' is preceded with whitespace.'
```

Comment:
The formatter originally formatted the curly braces to be `{}` but our config was complaining for it so I had to convert it to `{ }`.

I guess this is a false positive in our config, we should not be giving violation for `{}`. 

Should I revert back the braces to be `{}` and add violations for them? @romani

Reply:
I tried looking up to the style guide but it does not mentions that an empty curly braces should have whitespace between them. 

[4.6.2 Horizontal Whitespace](https://google.github.io/styleguide/javaguide.html#s4.6.2-horizontal-whitespace) section defines spacing rules, it is not mentioned anywhere that an empty pair of curly braces should have whitespace between them.

Reply:
opened an issue for this: https://github.com/checkstyle/checkstyle/issues/15338

converted `{ }` to `{}` and added their corresponding violation messages

Reply:
valid defect on Checkstyle side.

Reply:
keep violations, it is ok, after fix is provided it will be auto removed.